Homework-Mar-7/StackCalculator/StackCalculator/StackCalculator.cs

```diff
@@ -0,0 +1,106 @@
+namespace StackCalculator;
+
+using System.Text;
+
+/// <summary>
+/// Implements the execution of operations +, -, *, / on an arithmetic expression as a string in a postfix entry.
+/// </summary>
+/// <typeparam name="TStack">Stack implementation.</typeparam>
+public static class StackCalculator<TStack>
+    where TStack : IStack<double>, new()
+{
+    private static readonly Dictionary<string, Func<double, double, double>> Operations = new ()
+    {
+        { "+", (x, y) => x + y },
+        { "-", (x, y) => x - y },
+        { "*", (x, y) => x * y },
+        {
+            "/", (x, y) =>
+            {
+                if (Math.Abs(y) < 1e-10)
+                {
+                    throw new DivideByZeroException();
+                }
+
+                return x / y;
+            }
+        },
+    };
+
+    /// <summary>
+    /// Сalculates an arithmetic expression as a string in a postfix entry.
+    /// </summary>
+    /// <param name="expression">String in a postfix entry.</param>
+    /// <returns>Сalculation result.</returns>
+    /// <exception cref="InvalidOperationException">Invalid expression.</exception>
+    /// <exception cref="DivideByZeroException">You are trying to divide by zero.</exception>
+    public static double Calculate(string expression)
+    {
+        if (expression == string.Empty)
```

Comment:
Строки на нулл тоже советую проверять. Допустим, `Console.ReadLine()` может вполне возвращать `null`
```suggestion
        if (string.IsNullOrEmpty(expression))
```